lib/seahorn/BvOpSem2ExtraWideMemMgr.cc

```diff
@@ -583,7 +587,7 @@ ExtraWideMemManager<T>::memsetMetaData(ExtraWideMemManager::PtrTy ptr,
       MemoryFeatures::has_tracking(hana::type<T>{}),
       [&](auto _) {
         return _(m_main).memsetMetaData(ptr.getBase(), len, memIn.getRaw(),
-                                        val);
+                                        val = val);
```

Comment:
this assignment `val = val` looks off

Reply:
using named parameters just to make i know what i am passing since `len` and `val` have same datatype. Earlier due to automated refactoring, this was set to 0 and I had to visually squint to see that val was turned to 0

Reply:
I did not know that there are named parameters in C++. A comment would be useful. 

In other contexts, `v = v` evaluates to `v`, so code like `foo(v = v)` is legal in C, but it means `v = v; foo(v);`